### PR TITLE
Initial automated doctest, all current examples now pass, other documentation edits

### LINKED ISSUE #2012: [DOC] The html help for `multi_source_bfs` is not present in the online docs
`cugraph.multi_source_bfs` has a docstring, but it is not showing in the online docs.  It should be present under `traversal` with `bfs` : https://docs.rapids.ai/api/cugraph/stable/api_docs/traversal.html

### PR DESCRIPTION
**Pull request by betochimas**
Closes #2012 and #2022

### CODE REVIEW

File: python/cugraph/cugraph/dask/centrality/katz_centrality.py [hunk 1]
```diff
@@ -83,14 +84,17 @@ def katz_centrality(input_graph,
...
         A weight scalar - currently Not Supported
+
     max_iter : int
         The maximum number of iterations before an answer is returned. This can
         be used to limit the execution time and do an early exit before the
         solver reaches the convergence tolerance.
         If this value is lower or equal to 0 cuGraph will use the default
         value, which is 100.
+
     tolerance : float
```
Comment (rlratzel) [inline python/cugraph/cugraph/dask/centrality/katz_centrality.py:98]: This one is also `tol`

File: python/cugraph/cugraph/dask/centrality/katz_centrality.py [hunk 2]
```diff
@@ -99,6 +103,7 @@ def katz_centrality(input_graph,
         Setting too small a tolerance can lead to non-convergence due to
         numerical roundoff. Usually values between 1e-2 and 1e-6 are
         acceptable.
+
```
Comment (rlratzel) [inline python/cugraph/cugraph/dask/centrality/katz_centrality.py:106]: This line is fine but there should also be a blank line above `normalized` below.

File: python/cugraph/cugraph/link_analysis/hits.py [hunk 3]
```diff
@@ -38,17 +38,21 @@ def hits(G, max_iter=100, tol=1.0e-5, nstart=None, normalized=True):
...
-    max_iter : int
+
+    max_iter : int, optional (default=100)
         The maximum number of iterations before an answer is returned.
         The gunrock implementation does not currently support tolerance,
         so this will in fact be the number of iterations the HITS algorithm
         executes.
-    tolerance : float
+
+    tolerance : float, optional (default=1.0e-5)
```
Comment (rlratzel) [inline python/cugraph/cugraph/link_analysis/hits.py:48]: should be `tol`

File: python/cugraph/cugraph/linear_assignment/lap.py [hunk 4]
```diff
@@ -65,12 +65,12 @@ def hungarian(G, workers, epsilon=None):
...
-    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
-    >>> G = cugraph.Graph()
-    >>> G.from_cudf_edgelist(M, source='0', destination='1', edge_attr='2')
-    >>> cost, df = cugraph.hungarian(G, workers)
+    >>> # Download dataset from https://github.com/rapidsai/cugraph/datasets/..
+    >>> #  M = cudf.read_csv(datasets_path / 'bipartite.csv', delimiter=' ',
+    >>> #                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> # G = cugraph.Graph()
+    >>> # G.from_cudf_edgelist(M, source='0', destination='1', edge_attr='2')
+    >>> # cost, df = cugraph.hungarian(G, workers)
```
Comment (rlratzel) [inline python/cugraph/cugraph/linear_assignment/lap.py:73]: Can you add this FIXME after the docstring?
```
# FIXME: create bipartite.csv and uncomment out the above example
```

File: python/cugraph/cugraph/proto/components/scc.py [hunk 5]
```diff
@@ -55,12 +54,12 @@ def strong_connected_component(source, destination):
...
     Examples
     --------
-    >>> M = read_mtx_file(graph_file)
-    >>> sources = cudf.Series(M.row)
-    >>> destinations = cudf.Series(M.col)
+    # M = read_mtx_file(graph_file)
+    # sources = cudf.Series(M.row)
+    # destinations = cudf.Series(M.col)
+    # components, single_components, count =
+    #   cugraph.strong_connected_component(source, destination)
```
Comment (rlratzel) [inline python/cugraph/cugraph/proto/components/scc.py:61]: Need a FIXME for this example too.

File: python/cugraph/cugraph/structure/convert_matrix.py [hunk 6]
```diff
@@ -212,18 +229,23 @@ def from_pandas_edgelist(df,
...
+    destination : str or array-like, optional (default='destination')
         destination column name or array of column names
-    edge_attr : str or None
-        the weights column name. Default is None
-    renumber : bool
+
+    edge_attr : str or None, optional (default=None)
+        the weights column name.
+
+    renumber : bool, optional (default=True)
```
Comment (rlratzel) [inline python/cugraph/cugraph/structure/convert_matrix.py:244]: minor: `renumber` comes after `create_using` below in the arg list.

File: python/cugraph/cugraph/structure/symmetrize.py [hunk 7]
```diff
@@ -110,30 +115,29 @@ def symmetrize_ddf(df, src_name, dst_name, weight_name=None):
...
+    >>> # from cugraph.structure.symmetrize import symmetrize_ddf
+    >>> # Init a DASK Cluster
+    >>> # Download dataset from https://github.com/rapidsai/cugraph/datasets/..
+    >>> # chunksize = dcg.get_chunksize(datasets / 'karate.csv')
+    >>> # ddf = dask_cudf.read_csv(datasets/'karate.csv', chunksize=chunksize,
+    >>> #                          delimiter=' ',
+    >>> #                          names=['src', 'dst', 'weight'],
+    >>> #                          dtype=['int32', 'int32', 'float32'])
+    >>> # sym_ddf = symmetrize_ddf(ddf, "src", "dst", "weight")
+
```
Comment (rlratzel) [inline python/cugraph/cugraph/structure/symmetrize.py:140]: FIXME to uncomment this example

File: python/cugraph/cugraph/tree/minimum_spanning_tree.py [hunk 8]
```diff
@@ -87,6 +87,14 @@ def minimum_spanning_tree(
         A graph descriptor with a minimum spanning tree or forest.
         The networkx graph will not have all attributes copied over
 
+    Examples
+    --------
+    >>> M = cudf.read_csv(datasets_path / 'netscience.csv', delimiter='\t',
+    ...                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(M, source='0', destination='1')
+    >>> # cugraph.minimum_spanning_tree(G)
```
Comment (rlratzel) [inline python/cugraph/cugraph/tree/minimum_spanning_tree.py:96]: Is this and the algo call in the example below broken, or just an oversight?  If this isn't working in doctest, can you add a FIXME? If the algos themselves are broken, can you file an issue?

**Review by rlratzel — CHANGES_REQUESTED** [review]
Looks good overall, thanks for all the hard work of going through everything.
I found a few issues, which shouldn't be too hard to address.

File: python/cugraph/cugraph/generators/rmat.py [hunk 9]
```diff
@@ -280,7 +280,7 @@ def rmat(
         and duplicated edges.  Default is cugraph.DiGraph.
         NOTE: only the cugraph.DiGraph type is supported for multi-GPU
 
-    mg : bool
+    mg : bool, optional
```
Comment (acostadon) [inline python/cugraph/cugraph/generators/rmat.py:283]: should this have a default?

File: python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py [hunk 10]
```diff
@@ -366,9 +371,9 @@ def to_undirected(self, G):
             (u,v,weights) replaced by an undirected edge (u,v,weights).
         Examples
         --------
-        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
-        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
-        >>> DiG = cugraph.DiGraph()
+        >>> M = cudf.read_csv(datasets / 'karate.csv', delimiter=' ',
```
Comment (acostadon) [inline python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py:374]: datasets path here?

File: python/cugraph/cugraph/structure/symmetrize.py [hunk 11]
```diff
@@ -40,24 +40,29 @@ def symmetrize_df(df, src_name, dst_name, multi=False, symmetrize=True):
...
     dst_name : string
         Name of the column in the data frame containing the destination ids
-    multi : bool
+
+    multi : bool, optional (default=False)
         Set to True if graph is a Multi(Di)Graph. This allows multiple
         edges instead of dropping them.
-    symmetrize : bool
+
+    symmetrize : bool, optional
```
Comment (acostadon) [inline python/cugraph/cugraph/structure/symmetrize.py:54]: Needs default

File: python/cugraph/cugraph/structure/symmetrize.py (hunk 7 above)
Comment (acostadon) [reply python/cugraph/cugraph/structure/symmetrize.py:140]: do these need datasets_path?

**Review by acostadon — APPROVED** [review]
I think I found a few defaults missing. Hopefully my comments got recorded.
So many files...things look great now.

**Review by rlratzel — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**